### DISCUSSION TITLE
Error with volume rendering of dicom files when done with needle

### DISCUSSION THREAD

**Discussion by Karthikarc115** [needle-tools/needle-engine-support#137] [Export: Unity]
Hello, my name is Karthika, and I'm new to Unity WebGL and Needle. As part of my ongoing project, I had to do volume rendering of medical dicom slices for a project that should be worked on for the Web so that anyone can access it globally. I had completed the coding part and other functionalities that work perfectly for the Unity editor version and Windows standalone version when I tried to switch the platform in WebGL, I got an error with the GUI elements that I'm using in the project for local file browsing so that the user can upload the files from any system.

So, in order to ensure that my project runs smoothly, I tried to implement the needle engine. However, when I did so, I was only able to generate a cube in the output screen instead of the output that I had previously received from the normal webgl output. Any assistance would be greatly appreciated.

Here iam attaching the screenshot of the outputs that i had received in both unity webgl and needle engine output

![VR1](https://user-images.githubusercontent.com/99883381/228786849-d6de066f-c88d-44a5-895c-ac6dee517a48.png)
![1](https://user-images.githubusercontent.com/99883381/228787024-8bcd6da7-65bb-4ba1-bade-6e95272f06ad.PNG)
![2](https://user-images.githubusercontent.com/99883381/228787038-d2a6ac90-76ea-46df-95d5-9760aa09563e.PNG)
![3](https://user-images.githubusercontent.com/99883381/228787044-8c450b2c-7793-41c6-973d-1604b81911f4.PNG)
![Needle](https://user-images.githubusercontent.com/99883381/228787091-ea792d33-97a8-4353-8dab-a98e1b055943.PNG)

**Comment by marwie** [comment]
Hello, 

what is the file format that you use for your volumetric scans?

Here are some links on how to render/display volumetric data in threejs:
https://threejs.org/examples/webgl2_materials_texture3d.html
https://github.com/mrdoob/three.js/blob/dev/examples/webgl2_materials_texture3d.html

**Reply by Karthikarc115** [reply]
> Hello,
> 
> what is the file format that you use for your volumetric scans?
> 
> Here are some links on how to render/display volumetric data in threejs: https://threejs.org/examples/webgl2_materials_texture3d.html https://github.com/mrdoob/three.js/blob/dev/examples/webgl2_materials_texture3d.html

IMA,dcm extention files are used
